drivers/firmware/scmi/Kconfig

```diff
@@ -0,0 +1,52 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+if ARM_SCMI
+
+config ARM_SCMI_MAILBOX_TRANSPORT
+	bool "SCMI transport based on shared memory and doorbells"
+	default y
+	depends on DT_HAS_ARM_SCMI_ENABLED
+	select ARM_SCMI_TRANSPORT_HAS_STATIC_CHANNELS
+	help
+	  Enable support for SCMI transport based on shared memory
+	  and doorbells.
+
+config ARM_SCMI_TRANSPORT_HAS_STATIC_CHANNELS
+	bool "Transport layer has static channels"
+	default n
+	help
+	  Enable this if the SCMI transport layer uses static channels.
+	  What this is means is that the transport layer will place all
+	  of its channels in a special section from which the protocols
+	  will be able to obtain references. This is option is recommended
+	  for transport layers which don't require a complicated channel
+	  allocation scheme.
+
+config ARM_SCMI_SHMEM_INIT_PRIORITY
+	int "SCMI shared memory (SHMEM) initialization priority"
+	default 1
```

Comment:
Should we increase the priority value, for example, 4?
Before the SHMEM, there may be other dependencies. For example,
- MU (hardware message unit)
- GIC controller which MU may use.
Thanks.

Reply:
Good question. This is indeed an issue so it might be worth opening a discussion. Although, the SHMEM component is not the problem here as it doesn't have a dependency. The problematic components are the transport layer driver and the core (since this is initialized together with the transport layer).

For this particular case (i.e: 95) we're going to end up having to modify the initialization priorities of **some** components (SCMI or non-SCMI components) no matter what priorities we choose for the SCMI components (i.e: SHMEM and TRANSPORT).

As such, I agree with your proposal to increase the priority number (i.e: decrease the init priority) to allow for a bit more leeway when configuring the priorities of other components. This should be done for both the `SHMEM` and `TRANSPORT` components even though `SHMEM` has no dependency but should ideally have the same initialization priority as `MBOX` as they both make up the SCMI channels.